azure-devops/azext_devops/dev/pipelines/pipeline.py

```diff
@@ -116,26 +121,38 @@ def pipeline_run(id=None, branch=None, commit_id=None, name=None, open=False, va
     :type open: bool
     :param detect: Automatically detect organization and project. Default is "on".
     :type detect: str
+    :param parameters: json string for the variables you would like to set. Example: "{'name': 'value'}"
+    :type detect: str
     """
     organization, project = resolve_instance_and_project(
         detect=detect, organization=organization, project=project)
     if id is None and name is None:
         raise ValueError('Either the --id argument or the --name argument ' +
                          'must be supplied for this command.')
     client = get_build_client(organization)
+
     if id is None:
         id = get_definition_id_from_name(name, client, project, folder_path)
     definition_reference = DefinitionReference(id=id)
     branch = resolve_git_ref_heads(branch)
-    build = Build(definition=definition_reference, source_branch=branch, source_version=commit_id)
+    build = Build(definition=definition_reference,
+                  source_branch=branch, source_version=commit_id)
     if variables is not None and variables:
         build.parameters = {}
         for variable in variables:
             separator_pos = variable.find('=')
             if separator_pos >= 0:
-                build.parameters[variable[:separator_pos]] = variable[separator_pos + 1:]
+                build.parameters[variable[:separator_pos]
+                                 ] = variable[separator_pos + 1:]
             else:
-                raise ValueError('The --variables argument should consist of space separated "name=value" pairs.')
+                raise ValueError(
+                    'The --variables argument should consist of space separated "name=value" pairs.')
+    if parameters is not None and parameters:
+        parms = ast.literal_eval(parameters)
+        if build.parameters is None:
+            build.parameters = {}
+            for key, value in parms.items():
+                build.parameters[key] = value
```

Comment:
I don't think this is the ask, have asked feature requestor to confirm
https://github.com/Azure/azure-devops-cli-extension/issues/1110#issuecomment-880627929

I believe this has more to do with [parameters](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops&tabs=script)

Reply:
@roshan-sy : looks like we will have to keep this on hold because this would need client version upgrade.
https://github.com/Azure/azure-devops-cli-extension/issues/1110#issuecomment-880656006

Reply:
Noted